frontend-react/docs/test-conventions.md

```diff
@@ -0,0 +1,492 @@
+            <button type="submit">
+        </form>
+    );
+}
+```
+
+> Why?
+>
+> Testing behavior keeps tests agnostic to the implementation, which makes them more resistant to changes and refactors.  Ultimately, changing implementation shouldn't automatically imply needing to update tests unless the behavior is also changing.
+
+Somewhat of an exception to the aforementioned rule is pure functions, whose behavior is essentially its implementation:
+
+```tsx
+// source code
+function capitalize(str?: string) {
+    if (!str) {
+        return "";
+    }
+    
+    return str.charAt(0).toUpperCase() + str.substring(1).toLowerCase();
+}
+
+// tests
+describe('capitalizeString', () => {
+    describe("when the provided string is falsy", () => {
+        test("returns an empty string", () => {
+            expect(capitalizeString()).toEqual("");
+            expect(capitalizeString("")).toEqual("");
+        });
+    });
+
+    describe("when the provided string is truthy", () => {
+        test("returns the string with the first character capitalized (if applicable)", () => {
+            expect(capitalizeString("a")).toEqual("A");
+            expect(capitalizeString("A")).toEqual("A");
+			
+            expect(capitalizeString("abc")).toEqual("Abc");
+            expect(capitalizeString("Abc")).toEqual("Abc");
+            expect(capitalizeString("ABc")).toEqual("Abc");
+            expect(capitalizeString("ABC")).toEqual("Abc");
+    
+            expect(capitalizeString("1")).toEqual("1");
+            expect(capitalizeString("1abc")).toEqual("1abc");
+    
+            // and so on...
+        });
+    });
+});
+```
+
+### Tests should be as explicit as possible
+
+```tsx
+// bad
+expect(sum(1, 2, 3)).toEqual(1 + 2 + 3);
+
+const arrayA = [1, 2, 3];
+const arrayB = [4, 5, 6];
+const arrayC = [7, 8, 9];
+expect(
+    concatenate(arrayA, arrayB, arrayC)
+).toEqual([
+    arrayA,
+    arrayB,
+    arrayC
+].flat());
+       
+// good
+expect((sum(1, 2, 3))).toEqual(6);
+expect(
+    concatenate(arrayA, arrayB, arrayC)
+).toEqual([1, 2, 3, 4, 5, 6, 7, 8, 9]);
+```
+
+> Why?
+>
+> Tests should read as superficially clearly as possible, and they shouldn't reimplement the underlying logic.
+
+### Test coverage percentage is a barometer, not a doctrine
+
+> Why?
+>
+> It's still a good idea to keep an eye on the test coverage percentage as loose guidance (70% - 80%) as a metric for confidence in our code, but 100% coverage is a fallacy.  Especially on the client side where much of our code will be fed through a renderer verbatim, it's less important to attain "complete" coverage, and it may encourage an anti-pattern of introducing tautological tests.
+
+Consider the following example:
+
+```tsx
+// given an array of words, return a string that formats them into a human-readable list  
+// ex) ["apple", "banana", "cantaloupe", "durian"] ==> "apple, banana, cantaloupe, and durian"  
+function toSentence(words: string[]) {
```

Comment:
There's a really good existing example case here with the react-query hook's return values (like isLoading) not needing to be tested.